generators/server/templates/src/main/kotlin/package/security/oauth2/AuthorizationHeaderUtil.kt.ejs

```diff
@@ -18,31 +18,140 @@
 -%>
 package <%=packageName%>.security.oauth2
 
-import org.springframework.security.authentication.AbstractAuthenticationToken
+import org.slf4j.LoggerFactory
+import org.springframework.boot.web.client.RestTemplateBuilder
+import org.springframework.http.MediaType
+import org.springframework.http.RequestEntity
+import org.springframework.http.converter.FormHttpMessageConverter
 import org.springframework.security.core.context.SecurityContextHolder
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClient
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
 import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken
+import org.springframework.security.oauth2.client.http.OAuth2ErrorResponseErrorHandler
+import org.springframework.security.oauth2.core.*
+import org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames
+import org.springframework.security.oauth2.core.http.converter.OAuth2AccessTokenResponseHttpMessageConverter
 import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken
 import org.springframework.stereotype.Component
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.client.RestTemplate
+import java.net.URI
+import java.time.Duration
+import java.time.Instant
+import java.util.*
+import java.util.regex.Pattern
+
 
 @Component
-class AuthorizationHeaderUtil(private val clientService: OAuth2AuthorizedClientService) {
+class AuthorizationHeaderUtil(
+    private val clientService: OAuth2AuthorizedClientService,
+    private val restTemplateBuilder: RestTemplateBuilder
+) {
+
+    private val log = LoggerFactory.getLogger(javaClass)
+
+
+    fun getAuthorizationHeader(): Optional<String> {
```

Comment:
It would be nicer/cleaner to use `String?` as the return type instead of `Optional<String>`